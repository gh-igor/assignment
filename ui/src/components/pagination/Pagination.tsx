import React from "react";

type Props = {
    currentPage: number;
    itemsPerPageMax: number;
    itemsLength?: number;
    onPreviousClick: () => void;
    onNextClick: () => void;
};

const Pagination = ({ currentPage, itemsPerPageMax, itemsLength, onPreviousClick, onNextClick }: Props) => {
    return Number(itemsLength) > 0 ? (
        <>
            {currentPage > 1 && <button onClick={onPreviousClick}>&lt; previous</button>}
            <span> -= {currentPage} =- </span>
            {itemsLength === itemsPerPageMax && <button onClick={onNextClick}>next &gt;</button>}
        </>
    ) : null;
};

export default Pagination;
