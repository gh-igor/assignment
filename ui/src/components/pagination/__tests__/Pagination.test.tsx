import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Pagination from "../Pagination";

describe("renders Pagination", () => {

    test("first page", () => {
        const testProps = {
            currentPage: 1,
            itemsPerPageMax: 10,
            itemsLength: 10,
            onPreviousClick: jest.fn(),
            onNextClick: jest.fn(),
        };

        render(<Pagination {...testProps} />);

        const prevBtn = screen.queryByText(/previous/);
        expect(prevBtn).toBeNull();

        const nextBtn = screen.queryByText(/next/);
        expect(nextBtn).toBeInTheDocument();

        const pageNumber = screen.queryByText("-= 1 =-");
        expect(pageNumber).toBeInTheDocument();
    });

    test("next page", () => {
        const testProps = {
            currentPage: 2,
            itemsPerPageMax: 10,
            itemsLength: 10,
            onPreviousClick: jest.fn(),
            onNextClick: jest.fn(),
        };

        render(<Pagination {...testProps} />);

        const prevBtn = screen.queryByText(/previous/);
        expect(prevBtn).toBeInTheDocument();

        const nextBtn = screen.queryByText(/next/);
        expect(nextBtn).toBeInTheDocument();

        const pageNumber = screen.queryByText("-= 2 =-");
        expect(pageNumber).toBeInTheDocument();
    });

    test("last page", () => {
        const testProps = {
            currentPage: 3,
            itemsPerPageMax: 10,
            itemsLength: 9,
            onPreviousClick: jest.fn(),
            onNextClick: jest.fn(),
        };

        render(<Pagination {...testProps} />);

        const prevBtn = screen.queryByText(/previous/);
        expect(prevBtn).toBeInTheDocument();

        const nextBtn = screen.queryByText(/next/);
        expect(nextBtn).toBeNull();

        const pageNumber = screen.queryByText("-= 3 =-");
        expect(pageNumber).toBeInTheDocument();
    });

    test("matches snapshot", () => {
        const testProps = {
            currentPage: 3,
            itemsPerPageMax: 10,
            itemsLength: 9,
            onPreviousClick: jest.fn(),
            onNextClick: jest.fn(),
        };

        const tree = renderer.create(<Pagination {...testProps} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test("btn clickable", () => {
        const testProps = {
            currentPage: 2,
            itemsPerPageMax: 10,
            itemsLength: 10,
            onPreviousClick: jest.fn(),
            onNextClick: jest.fn(),
        };

        render(<Pagination {...testProps} />);

        fireEvent.click(screen.getByText(/next/i));
        fireEvent.click(screen.getByText(/previous/i));

        expect(testProps.onNextClick).toHaveBeenCalled();
        expect(testProps.onPreviousClick).toHaveBeenCalled();
    });
});
