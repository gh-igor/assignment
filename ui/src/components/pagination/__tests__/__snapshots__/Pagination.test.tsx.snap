// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Pagination matches snapshot 1`] = `
Array [
  <button
    onClick={[MockFunction]}
  >
    &lt; previous
  </button>,
  <span>
     -= 
    3
     =- 
  </span>,
]
`;
